R/linting.R

```diff
@@ -0,0 +1,48 @@
+#' Use code linting
+#'
+#' `use_lintr()` Adds support for code linting using the `lintr` package in one of three ways:
+#' local interactive linting only, non-failing (post-build) Travis CI linting, or failing local and Travis CI `testthat` unit test.
+#'
+#' @param type Options are \code{local}, \code{travis} or \code{test}.
+#' @export
+use_lintr <- function(type = c("local", "travis", "test")){
+  type <- match.arg(type)
+
+  if(type == "test"){
+    dir.create("inst", showWarnings = FALSE)
+    .lintr <- "inst/.lintr"
+  } else {
+    .lintr <- ".lintr"
+  }
+
+  sink(.lintr)
+  cat(paste0("linters: with_defaults(\n  ", "line_length_linter(80))\n"))
+  sink()
+
+  if(type == "test" && !file.exists(".lintr")) file.symlink("inst/.lintr", ".lintr")
```

Comment:
This symbolic link approach is problematic. This is the "old way" of making lintr used as a `testthat` unit test work with Travis CI. I think there is a newer, better way to do this that does not involve a symbolic link to a `.lintr` file located inside `inst` but I do not know how this should be altered to get rid of the linking approach for the unit test setup.

Reply:
usethis uses the fs package for all file system and file path operations. So that should be on the radar as you resolve this.